airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_read_processor.py

```diff
@@ -56,6 +56,25 @@ def __init__(
         self._message_repository = message_repository
         self._partition_reader = partition_reader
 
+    def on_stream_availability(self, sentinel: StreamAndStreamAvailability) -> Optional[AirbyteMessage]:
+        stream = sentinel.stream
+        if sentinel.availability.is_available():
+            self._logger.info(f"Marking stream {stream.name} as STARTED")
+            self._logger.info(f"Syncing stream: {stream.name} ")
+            return stream_status_as_airbyte_message(
+                stream.as_airbyte_stream(),
+                AirbyteStreamStatus.STARTED,
+            )
+        else:
+            # Remove the stream from _streams_currently_generating_partitions because no partitions will be generated
+            self._streams_currently_generating_partitions.remove(stream.name)
+            self._logger.warning(
+                f"Skipped syncing stream '{sentinel.stream.name}' because it was unavailable. {sentinel.availability.message()}"
+            )
+            if self._stream_instances_to_start_partition_generation:
+                self.start_next_partition_generator()
+            return None
+
```

Comment:
This overhead is due to the fact that we're submitting the partition generating tasks in separate stages so that the workers aren't blocked on I/O, right? Assuming so I think its another argument in favor of revisiting our approach, per [my comment in the other PR](https://github.com/airbytehq/airbyte/pull/32411/files/2ebfbf8f829d52a076fcd5d907974672b1483975#r1397964110).